### ISSUE TITLE
All devices stopped suddenly responding

### ISSUE THREAD

**Issue by ziggitipop** [tuya/tuya-homebridge#299]
Several light switches suddenly stopped responding from homekit. Working fine within the tuya app but no response at all through homebridge.

**Comment by kbyquist** [comment]
I'm having the same issue. I can control through the Tuya app, but homebridge is not sending any request to Tuya when an outlet is turned on in HomeKit.

**Comment by AlexKvam** [comment]
I'm having the same issue here are my logs:
`[10/13/2022, 7:23:36 PM] Homebridge v1.4.1 (HAP v0.10.2) (Homebridge 75D7) is running on port 51635.
TuyaOpenAPI response: {"code":2406,"msg":"skill id invalid","success":false,"t":1665681816231,"tid":"c524cb704b1b11edb8a1ca0f69d78a9e"} path = /v1.0/iot-01/associated-users/actions/authorized-login
TypeError: Cannot destructure property 'access_token' of 'res.result' as it is undefined.
    at TuyaSHOpenAPI._refreshAccessTokenIfNeed (/usr/local/lib/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:49:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at TuyaSHOpenAPI.request (/usr/local/lib/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:145:7)
    at TuyaSHOpenAPI.getDevices (/usr/local/lib/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:63:15)
    at TuyaPlatform.initTuyaSDK (/usr/local/lib/node_modules/homebridge-tuya-platform/index.js:91:19)
Attention⚠️ ⚠️ ⚠️ ! You get an error!
Failed to get device information. Please check if the config.json is correct.`

**Comment by RidasBuzius** [comment]
Same error

**Comment by Kohle81** [comment]
Same error, too.

**Comment by FarisDIL** [comment]
Same issue :|

**Comment by cjstone** [comment]
Same!!  @zlinoliver, any updates?

**Comment by cjstone** [comment]
@Hanh94 ?

**Comment by Azukovskij** [comment]
Same as https://github.com/tuya/tuya-homebridge/issues/298, I've added a [workaround|https://github.com/tuya/tuya-homebridge/issues/298#issuecomment-1278238870 ] to resolve this for my smart home

**Comment by cjstone** [comment]
Thanks @Azukovskij! Works great now!

**Comment by timhbf** [comment]
May I know how to apply this workaround? Thanks

**Comment by freemanuk** [comment]
Facing the same problem.

**Comment by Azukovskij** [comment]
> May I know how to apply this workaround? Thanks

@timhbf you need to modify file from an error ‘ /usr/local/lib/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js’ and add changes of linked commit.

**Comment by mmaslen** [comment]
Tuya are now asking me to extend my trial period - is that normal - i'm not even using this for development, just a home user - if they are going to make me pay for the service now i'll reconsider me using their devices, not a good look 👎

**Comment by antonkrall** [comment]
> /usr/local/lib/node_modules/homebridge-tuya-platform/lib

Can confirm that it works!!!

**Comment by antonkrall** [comment]
Why wouldnt the creator of the plugin just apply this patch and push it?

**Comment by freemanuk** [comment]
I log in to Tuya iot platform, under Device option and switched it back to Central Europe Data Center and got back my device list. Everything works now. I don't know why is was switched to China Data Center.

**Comment by Kohle81** [comment]
I don‘t know what to do?

**Comment by preethaml7** [comment]
Happened to me today. All I did was unlink the app from the tuya cloud website, and re-link it again by scanning the QR code shown on tuya cloud with the tuya app. (provide read and write permission to devices in the tuya cloud prompt after scanning the QR code)
Reload the integration in Homebridge and voilà, everything should be working as before.
If you are still facing issues, go to home bridge settings Remove All Cached Accessories and try the above step hopefully you should get it working too.

**Comment by mmaslen** [comment]
> Happened to me today. All I did was unlink the app from the tuya cloud website, and re-link it again by scanning the QR code shown on tuya cloud with the tuya app. (provide read and write permission to devices in the tuya cloud prompt after scanning the QR code) Reload the integration in Homebridge and voilà, everything should be working as before. If you are still facing issues, go to home bridge settings Remove All Cached Accessories and try the above step hopefully you should get it working too.

tried that, no success

**Comment by PNWOregon** [comment]
My devices have been unusable In HomeBridge for 2 days now. I’ve tried several of the suggested fixes except for Azukovskij‘a “work around” since I don’t under stand it. Hopefully this gets resolved soon.

**Comment by mmaslen** [comment]
ok, it's finally working for me with the standard plug-in - I ended up changing and ensuring that the iOS app password, the IOT platform password and the plugin password were all identical

**Comment by DavKirsch** [comment]
I had this same issue and just solved it today. About a month ago, all my devices on the Tuya Platform bridge stopped working in Homekit. They were still working on the Smart Life app though...

After a lot of searching, I discovered the IoT Core Services on the Tuya IoT Developer Platform had been working via a free trial and that had expired. There was an orange button to request an extension of the trial (as buying the bundle was $25K+). I asked for an extension for 6 months for my Hoobs Smart Hub, which was granted by the next morning. Then, I restarted the Bridge in HOOBs and all my devices were cached again and working in Homekit!! Took me a month to solve but everything is up and running.

FYI, You will need to repeat this process every 6 months though, the only downside.

[Workflow.with.Tuya.pdf](https://github.com/tuya/tuya-homebridge/files/10296130/Workflow.with.Tuya.pdf)
